### PR TITLE
force-ir flag on HailContext

### PR DESCRIPTION
**Pull request by catoverdrive**

### CODE REVIEW

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 1]
```diff
@@ -472,6 +472,39 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
...
+        ast.`type`.asInstanceOf[TStruct].size < 500 &&
+          globalType.size < 3 &&
+          colType.size == 1 &&
+          Set(TInt32(), +TInt32(), TString(), +TString())
+            .contains(colType.types(0))
+
+      case this.colAxis =>
+        ast.`type`.asInstanceOf[TStruct].size < 500
+
+      case this.entryAxis =>
```
Comment (danking) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: You can use `case this.entryAxis | this.rowAxis =>` if you want to match different things

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 2]
```diff
@@ -472,6 +472,39 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
...
+    axis match {
+      case this.globalAxis =>
+        ast.`type`.asInstanceOf[TStruct].size < 500
+
+      case this.rowAxis =>
+        ast.`type`.asInstanceOf[TStruct].size < 500 &&
+          globalType.size < 3 &&
+          colType.size == 1 &&
+          Set(TInt32(), +TInt32(), TString(), +TString())
+            .contains(colType.types(0))
```
Comment (danking) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: what's the reasoning for this?

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 3]
```diff
@@ -472,6 +472,39 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
         matrixType,
         MatrixValue(matrixType, globals, colValues, rvd)))
 
+  private type Axis = Int
+  private val globalAxis: Axis = 0
+  private val rowAxis: Axis = 1
+  private val colAxis: Axis = 2
+  private val entryAxis: Axis = 3
```
Comment (danking) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: generally I prefer `private[this] ` whenever possible (there's very minor performance improvements for field accesses [b/c/ there's no indirection])

File: src/main/scala/is/hail/variant/MatrixTable.scala (hunk 2 above)
Comment (catoverdrive) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: I think this covers a lot of common cases where you've got either just a string col ID or you've created something from an index/range. Mostly idk, I didn't want it to just check that there was only one field but have that field be something huge, but I don't have super strong feels about keeping this.

**Pull request merged**